app/feature/feature-insurances/src/main/kotlin/com/hedvig/android/feature/insurances/insurancedetail/ContractDetailViewModel.kt

```diff
@@ -5,23 +5,27 @@ import androidx.lifecycle.viewModelScope
 import arrow.core.raise.either
 import com.hedvig.android.core.common.RetryChannel
 import com.hedvig.android.core.demomode.Provider
-import com.hedvig.android.feature.insurances.insurancedetail.data.ContractDetails
-import com.hedvig.android.feature.insurances.insurancedetail.data.GetContractDetailsUseCase
+import com.hedvig.android.feature.insurances.data.GetInsuranceContractsUseCase
+import com.hedvig.android.feature.insurances.data.InsuranceContract
+import kotlin.time.Duration.Companion.seconds
 import kotlinx.coroutines.flow.SharingStarted
 import kotlinx.coroutines.flow.StateFlow
 import kotlinx.coroutines.flow.WhileSubscribed
 import kotlinx.coroutines.flow.stateIn
-import kotlin.time.Duration.Companion.seconds
 
 internal class ContractDetailViewModel(
   contractId: String,
-  private val getContractDetailsUseCase: Provider<GetContractDetailsUseCase>,
+  private val getInsuranceContractsUseCase: Provider<GetInsuranceContractsUseCase>,
 ) : ViewModel() {
   private val retryChannel = RetryChannel()
   val uiState: StateFlow<ContractDetailsUiState> = retryChannel.transformLatest {
     emit(ContractDetailsUiState.Loading)
     val uiState = either {
-      getContractDetailsUseCase.provide().invoke(contractId).bind()
+      getInsuranceContractsUseCase
+        .provide()
+        .invoke()
+        .bind()
+        .first { it.id == contractId }
```

Comment:
Now I think this will throw an exception if the contract ID does not get found at all, since we do `first {}` on the List<> itself.
Could we perhaps do a `firstOrNull {}` and then do a [ensureNotNull](https://github.com/HedvigInsurance/android/blob/d2bff2658dad04316166214d4fe9e34a354af3cb/app/data/data-travel-certificate/data-travel-certificate-public/src/main/kotlin/com/hedvig/android/data/travelcertificate/GetTravelCertificateSpecificationsUseCase.kt#L44) and return a typed error instead, so it doesn't crash completely?